packages/graph/hash_graph/lib/graph/src/store/postgres/query/expression/join.rs

```diff
@@ -0,0 +1,117 @@
+use std::fmt;
+
+use crate::store::postgres::query::{Condition, Expression, Table, Transpile};
+
+#[derive(Debug, PartialEq, Eq, Hash)]
+pub struct JoinExpression<'q> {
+    pub table: Table,
+    pub on: Condition<'q>,
+}
+
+impl<'q> JoinExpression<'q> {
+    #[must_use]
+    pub const fn from_tables(table: Table, on: Table) -> Self {
+        Self {
+            table,
+            on: Condition::Equal(
+                Some(Expression::Column(table.source_join_column())),
+                Some(Expression::Column(on.target_join_column())),
+            ),
+        }
+    }
+}
+
+impl Transpile for JoinExpression<'_> {
+    fn transpile(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        fmt.write_str("JOIN ")?;
+        if self.table.alias.is_some() {
+            let unaliased_table = Table {
+                name: self.table.name,
+                alias: None,
+            };
+            unaliased_table.transpile(fmt)?;
+            fmt.write_str(" AS ")?;
```

Comment:
Thanks for making the `AS` explicit! It's optional in Postgres and very confusing when not added :smile: